src/esp/assets/BaseMesh.h

```diff
@@ -19,49 +25,141 @@
 namespace esp {
 namespace assets {
 
+/**
+  @brief Enumeration of mesh types supported by the simulator. Each entry
+  suggests a derived class of @ref BaseMesh implementing the specific storage
+  and processing interface to accomodate differing asset formats. Identifies the
+  derived variant of this object.
+  */
 enum SupportedMeshType {
+  /**
+   * @brief Undefined mesh types are created programmatically without a specific
+   * format or loaded from an unkown format. Support for this type and behavior
+   * is likely limited. Object type is likely @ref BaseMesh.
+   */
   NOT_DEFINED = -1,
+
+  /**
+   * @brief Instance meshes loaded from sources including segmented object
+   * identifier data (e.g. semantic data: chair, table, etc...). Sources include
+   * .ply files and reconstructions of Matterport scans. Object is likely of
+   * type @ref GenericInstanceMeshData or Mp3dInstanceMeshData.
+   */
   INSTANCE_MESH = 0,
+
+  /**
+   * @brief Meshes loaded from Replica dataset. Object is likely type @ref
+   * PTexMeshData.
+   */
   PTEX_MESH = 1,
+
+  /**
+   * @brief Meshes loaded from gltf format (i.e. .glb file). Object is likely
+   * type @ref GltfMeshData.
+   */
   GLTF_MESH = 2,
+
+  /**
+   * @brief Number of enumerated supported types.
+   */
```

Comment:
Unlike everything else, with m.css enum values don't have brief docs as these would add too much noise to API listings. That means those get ignored and your enum values won't have any visible documentation. The script prints a warning if it encounters such issue, but I guess it'd be drowned among all other warnings right now anyway :)

(How to fix -- simply remove the `@brief`, leaving just plain paragraphs.)

(I'll stop complaining about `@brief` now, sorry :D)

Reply:
Looks like they don't get briefs as you said, but at least these are converted automatically into paragraphs. 🤔 I guess it makes sense to remove brief for specific enumerators but leave it for the enum type. 👍 

![image](https://user-images.githubusercontent.com/1445143/71704641-51415c80-2d90-11ea-8c8a-129ea8d5b21b.png)